roles/reproducer/molecule/job_uri/tasks/test_regex.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Test regex
+  block:
+    - name: "Call task file with {{ regex_test.string }}"
+      vars:
+        cifmw_job_uri: '{{ regex_test.string }}'
+      ansible.builtin.include_role:
+        name: reproducer
+        tasks_from: parse_cifmw_job_uri.yml
+
+    - name: Set fail variable if string incorrectly passes regex
```

Comment:
ok, so we expect to reach that step if the regex does match the string. There's an assert in the parse_cifmw_job_uri to ensure that.
Maybe add a small comment explaining it?

Reply:
I've added a comment to the top of this file to explain what's being tested, let me know if you would like more.

You can also test this locally with the following documented vars to see all scenarios in aciton:
```
  vars:
    regex_tests:
      - pass: true # This will pass if regex is right
        string: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0304e7/anything/////
        return: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0304e7
      - pass: false # This will pass if regex if working as it will expectadly fail
        string: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0
      - pass: false # This will fail as it will pass regex but we define it to fail
        string: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0304e7/anything/////
      - pass: true # This will fail as it will not pass regex but we expect it to pass
        string: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d03
        return: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0304e7
      - pass: true # This will fail because return value is wrong
        string: https://logserver.rdoproject.org/94/1494/1f72e2797961e0fb0192e3280c3027a8d70b4e45/github-check/cifmw-molecule-libvirt_manager/d0304e7
        return: https://logserver.rdoproject.org/94/1494/1f72e
```